### PR TITLE
Address API review comments

### PR DESCRIPTION
**Pull request by pakrym**
https://github.com/dotnet/apireviews/tree/master/2018/System.IO.Pipelines

1. Add `Slice(int, int)` overloads
2. Rename `Seek` to `GetPosition`
4. Rename `VirtualIndex` to `RunningLength`
5. Remove `ROB.ctor(OwnedMemory)` add `ROB.ctor(Memory)`

### CODE REVIEW

File: src/System.Buffers.Primitives/System/Buffers/ReadOnlyBuffer.cs [hunk 1]
```diff
@@ -146,50 +133,71 @@ public ReadOnlyBuffer(IEnumerable<Memory<T>> buffers)
...
+        /// <param name="offset">The index at which to begin this slice.</param>
+        /// <param name="length">The length of the slice</param>
+        public ReadOnlyBuffer<T> Slice(int offset, int length) => Slice((long)offset, length);
+
+        /// <summary>
+        /// Forms a slice out of the given <see cref="ReadOnlyBuffer"/>, beginning at 'start', ending at 'end' (inclusive).
+        /// </summary>
+        /// <param name="offset">The index at which to begin this slice.</param>
+        /// <param name="end">The end (inclusive) of the slice</param>
+        public ReadOnlyBuffer<T> Slice(int offset, Position end) => Slice((long)offset, end);
```
Comment (davidfowl) [inline src/System.Buffers.Primitives/System/Buffers/ReadOnlyBuffer.cs]: I think this defeats the purpose of the int overload.

Comment (pakrym) [reply src/System.Buffers.Primitives/System/Buffers/ReadOnlyBuffer.cs]: I was adding it to get surface to the place we want.

Comment (pakrym) [reply src/System.Buffers.Primitives/System/Buffers/ReadOnlyBuffer.cs]: Fixed.

**Comment by pakrym** [comment]
Any objection to merging this?

/cc @KrzysztofCwalina @davidfowl @joshfree @muratg

File: src/System.Buffers.Primitives/System/Buffers/ReadOnlyBuffer.cs [hunk 2]
```diff
@@ -146,46 +133,71 @@ public ReadOnlyBuffer(IEnumerable<Memory<T>> buffers)
         /// <summary>
         /// Forms a slice out of the given <see cref="ReadOnlyBuffer"/>, beginning at 'start', and is at most length bytes
```
Comment (sebastienros) [inline src/System.Buffers.Primitives/System/Buffers/ReadOnlyBuffer.cs]: Rename 'start' parameter

Comment (sebastienros) [reply src/System.Buffers.Primitives/System/Buffers/ReadOnlyBuffer.cs]: Not the only occurence btw.

**Comment by KrzysztofCwalina** [comment]
I like all the changes except IBufferList to IMemoryListNode. I remember a discussion about this, but I was not aware that we made a decision to go through with the rename. I think IMemoryListNode is a decent name from the perspective of the implementer, but not the consumer, and we usually optimize for consumers. The consumer does not care how the datastructure is implemented. In fact, IBufferList can be implemented without any nodes, i.e. as an array.

**Comment by pakrym** [comment]
They still look like nodes of the linked list with `data` and `next` no matter the implementation. What name do you like?

**Comment by KrzysztofCwalina** [comment]
I like IBufferList<T> or ISegmentList<T> with the Memory property being renamed to Segment.

**Comment by pakrym** [comment]
OK, I'll rename it back, for now, so everything else can be merged.

**Comment by KrzysztofCwalina** [comment]
looks good

**Pull request merged**